x-pack/plugins/alerts/server/lib/alert_execution_status.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AlertTaskState, AlertExecutionStatus, RawAlertExecutionStatus } from '../types';
+import { getReasonFromError } from './error_with_reason';
+
+export function executionStatusFromState(state: AlertTaskState): AlertExecutionStatus {
+  const instanceIds = Object.keys(state.alertInstances ?? {});
+  return {
+    date: new Date(),
+    status: instanceIds.length === 0 ? 'ok' : 'active',
+  };
+}
+
+export function executionStatusFromError(error: Error): AlertExecutionStatus {
+  return {
+    date: new Date(),
+    status: 'error',
+    error: {
+      reason: getReasonFromError(error),
+      message: error.message,
+    },
+  };
+}
+
+export function alertExecutionStatusToRaw({
+  date,
+  status,
+  error,
+}: AlertExecutionStatus): RawAlertExecutionStatus {
+  return {
+    date: date.toISOString(),
+    status,
+    error: error ?? null,
+  };
+}
+
+export function alertExecutionStatusFromRaw(
+  rawAlertExecutionStatus?: Partial<RawAlertExecutionStatus> | null | undefined
+): AlertExecutionStatus | undefined {
+  if (!rawAlertExecutionStatus) return undefined;
+
+  const { date, status = 'unknown', error } = rawAlertExecutionStatus;
+
+  let parsedDateMillis = date ? Date.parse(date) : Date.now();
+  if (isNaN(parsedDateMillis)) {
+    // TODO: log a message?
```

Comment:
but now I have to thread a Logger through a few methods :-(

Reply:
wow, this is going to get nasty - but TBH, these locations are a great place to have a logger, since we don't do much verification on the raw data anyway ...